libnss/src/host.rs

```diff
@@ -0,0 +1,232 @@
+use crate::interop::CBuffer;
+use std::collections::VecDeque;
+use std::mem;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+pub struct Host {
+    pub name: String,
+    pub aliases: Vec<String>,
+    pub addresses: Addresses,
+}
+
+pub enum Addresses {
+    V4(Vec<Ipv4Addr>),
+    V6(Vec<Ipv6Addr>),
+}
+
+impl Host {
+    pub unsafe fn to_c_hostent(self, hostent: *mut CHost, buffer: &mut CBuffer) {
+        (*hostent).name = buffer.write_str(self.name);
+        (*hostent).h_aliases = buffer.write_strs(&self.aliases);
+
+        let ptr_size = mem::size_of::<*mut libc::c_char>() as isize;
+
+        let (addr_len, count) = match &self.addresses {
```

Comment:
If I am understanding this correctly, this makes it possible to only return a list of V4 addresses OR a list of V6 addresses, but not possible to return a list of all V4 AND V6 addresses for a host.

Assuming my interpretation is correct, is this an inherit limitation of the NSS C API or just of this shim?

Reply:
I read the https://ftp.gnu.org/old-gnu/Manuals/glibc-2.2.3/html_chapter/libc_16.html#SEC318 link in these docs and it does appear to be inherent in the shape of the struct we need to return. Bummer.